activestorage/app/controllers/active_storage/blobs_controller.rb

```diff
@@ -0,0 +1,22 @@
+# Take a signed permanent reference for a blob and turn it into an expiring service URL for download.
```

Comment:
> Are they meant as superclasses to inherit from?

No, though you could probably extend them if you so desired. Apps use them by sending requests to them, either via routes or `DiskService#url{,_for_direct_upload}`.